modules/internal/ChapelAutoAggregation.chpl

```diff
@@ -0,0 +1,388 @@
+/*
+ * Copyright 2020-2021 Hewlett Packard Enterprise Development LP
+ * Copyright 2004-2019 Cray Inc.
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+module ChapelAutoAggregation {
+  private use CopyAggregation;
+  private use AggregationPrimitives only getEnvInt;
+
+  config param verboseAggregation = false;
+
+  private const yieldFrequency = getEnvInt("CHPL_AGGREGATION_YIELD_FREQUENCY", 1024);
+  private const dstBuffSize = getEnvInt("CHPL_AGGREGATION_DST_BUFF_SIZE", 4096);
+  private const srcBuffSize = getEnvInt("CHPL_AGGREGATION_SRC_BUFF_SIZE", 4096);
```

Comment:
Any reason to have these in ChapelAutoAggregation instead of CopyAggregation?

Reply:
Nothing too deep. They are more less part of the interface (for some definition of "interface") for automatic aggregation, where whatever's inside `CopyAggregation` is used by the compiler only. I have no problem putting them back in to `CopyAggregation`, if you thinkg that's better.

Reply:
I have a slight preference for moving to CopyAggregation, but not a very strong one so I don't mind if you leave as is

Reply:
Done :)